mlflow/sklearn/__init__.py

```diff
@@ -1335,7 +1462,9 @@ def patched_predict(original, self, *args, **kwargs):
         """
         status = _get_autologging_metrics_manager()
         if status.should_log_post_training_metrics() and status.get_run_id_for_model(self):
-            predict_result = original(self, *args, **kwargs)
+            # Avoid nested patch when nested inference calls happens.
+            with status.disable_log_post_training_metrics():
```

Comment:
Nice! Which test case covers this?

Reply:
covered in new added test `test_patch_for_delegated_method`

Reply:
Thank you!